out_writeapi.go

```diff
@@ -284,16 +284,22 @@ func sendRequestExactlyOnce(ctx context.Context, data [][]byte, config **outputC
 func sendRequestRetries(ctx context.Context, data [][]byte, config **outputConfig, streamIndex int) error {
 	retryer := newStatelessRetryer((*config).numRetries)
 	attempt := 0
-	currStream := (*(*config).managedStreamSlice)[streamIndex]
 	for {
 		err := sendRequestExactlyOnce(ctx, data, config, streamIndex)
 		if err == nil {
 			break
 		}
 		//unsuccesful data append
 		if rebuildPredicate(err) {
-			currStream.managedstream.Finalize(ctx)
-			currStream.managedstream.Close()
+			streamSlice := (*config).managedStreamSlice
+			// finalizes all streams
+			for i := 0; i < len(*streamSlice); i++ {
+				currStream := (*streamSlice)[i]
+				currStream.managedstream.Finalize(ctx)
+				currStream.managedstream.Close()
```

Comment:
Would it be possible to test that all streams are getting closed and with a new flush it will only re-create one stream?

Reply:
Would you like this in a unit test or just like a local test?

Reply:
Not sure what you mean by a local test